### PR TITLE
Fix the concurrent rebuild limit not working issue

### PR DESCRIPTION
**Pull request by shuo-wu**
longhorn/longhorn#2460

### CODE REVIEW

File: controller/replica_controller.go [hunk 1]
```diff
@@ -67,7 +67,8 @@ type ReplicaController struct {
 
 	instanceHandler *InstanceHandler
 
-	rebuildingLock *sync.Mutex
+	rebuildingLock          *sync.Mutex
+	inProgressRebuildingMap map[string]struct{}
```
Comment (joshimoo) [inline controller/replica_controller.go:71]: NIT: you can use map[string]interface as well then just push nil. If you only care about the key and not the value.
Both approaches are fine :)

File: controller/replica_controller.go [hunk 2]
```diff
@@ -98,7 +99,8 @@ func NewReplicaController(
 
 		ds: ds,
 
-		rebuildingLock: &sync.Mutex{},
+		rebuildingLock:          &sync.Mutex{},
+		inProgressRebuildingMap: map[string]struct{}{},
```
Comment (joshimoo) [inline controller/replica_controller.go:103]: NIT: I would probably use `make(map[string]ValueType)` instead. The inline initialization is nice if you immediately add values. But again not important :)

**Review by innobead — APPROVED** [review]
LGTM

**Comment by shuo-wu** [comment]
@joshimoo Thank you for your suggestions. I will revisit them next week.

**Pull request merged**